packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -21,6 +21,9 @@ import 'widget_inspector.dart';
 
 export 'dart:ui' show AppLifecycleState, Locale;
 
+/// A callback that can be registered with [WidgetsBinding.registerHotRestartCallback].
+typedef PreHotRestartCallback = FutureOr<void> Function();
```

Comment:
I don't usually like FutureOr, but in this case since we want to support `async` functions, the futureOr lets you pass tear-offs of `void` returning methods without wrapping in another function.

```
var foos = <int>[];
...
WidgetsBinding.instance!.registerHotRestartCallback(foos.clear);
// vs
WidgetsBinding.instance!.registerHotRestartCallback(() async => foos.clear());
```

Reply:
`Future<void>?` is probably better. Or just `Future<void>` since really it's trivial to add `async` to a function.

Reply:
`Object?` will also accept either a future returning function or a void function/tear-off, i'll switch to that

Reply:
done